wallet/src/de/schildbach/wallet/service/BlockchainServiceImpl.java

```diff
@@ -1481,4 +1490,103 @@ private void closeStream(InputStream mnlistinfoBootStrapStream) {
             }
         }
     }
+
+    // TODO: should we have a backup blockchain file?
+//    private NewBestBlockListener newBestBlockListener = block -> {
+//        try {
+//            backupBlockStore.put(block);
+//        } catch (BlockStoreException x) {
+//            throw new RuntimeException(x);
+//        }
+//    };
+
+    private boolean verifyBlockStore(BlockStore store) throws BlockStoreException {
+            StoredBlock cursor = store.getChainHead();
+            for (int i = 0; i < 10; ++i) {
+                cursor = cursor.getPrev(store);
+                if (cursor == null || cursor.getHeader().equals(Constants.NETWORK_PARAMETERS.getGenesisBlock())) {
+                    break;
+                }
+            }
+            return true;
+    }
+
+    private boolean verifyBlockStore(BlockStore store, ScheduledExecutorService scheduledExecutorService) {
+        try {
+            ScheduledFuture<Boolean> future = scheduledExecutorService.schedule(() -> verifyBlockStore(store), 100, TimeUnit.MILLISECONDS);
+            return future.get(1, TimeUnit.SECONDS);
+        } catch (Exception e) {
+            log.warn("verification of blockstore failed:", e);
+            return false;
+        }
+    }
+
+    // TODO: should we have a backup blockchain file?
+//    public static void copyFile(File source, File destination) throws IOException {
+//        try (FileChannel sourceChannel = new FileInputStream(source).getChannel();
+//             FileChannel destChannel = new FileOutputStream(destination).getChannel()) {
+//            sourceChannel.transferTo(0, sourceChannel.size(), destChannel);
+//        }
+//    }
```

Comment:
When we thought that the blockchain file was corrupt, I thought to back it up and have a means of restoring it.

I will leave this for now and think a bit more about it.  The backup file may not be necessary.